### DISCUSSION TITLE
Welcome to Acode Discussions!

### DISCUSSION THREAD

**Discussion by deadlyjack** [Acode-Foundation/Acode#443] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Akash-nath29** [comment]
Can't able to run Python in app...would like to get Python interpreter in app

**Reply by deadlyjack** [reply]
Install plugin to run python.

**Comment by ManuelPeh76** [comment]
Question: Is it possible, to somehow run Acode in the browser? Maybe with a different file handling?

**Reply by sebastianjnuwu** [reply]
Unfortunately not

**Comment by SmashRex** [comment]
Can anyone add a function in acode, when we click a namespace of path it opens that file

**Reply by bajrangCoder** [reply]
Added , Check out Path linker plugin

**Comment by JankleDev** [comment]
Heya

Here's some feature requests i have for y'all 🙂

https://github.com/deadlyjack/Acode/issues/669

**Comment by Medboy224** [comment]
Hello, is it possible to run C++ code in Acode ? I have try but it doesn't work.

**Reply by bajrangCoder** [reply]
Use termux with any Terminal plugin (like AcodeX)

**Reply by Medboy224** [reply]
Sorry I didn't understand can you explain please. How I can run c++ code in Acode

**Reply by bajrangCoder** [reply]
Download and setup any Terminal plugin or simply download Termux app from their GitHub page or f-droid and now you can use termux terminal like any Linux terminal where you can run any c++ code or other languages too. 
For c++ :
`gcc yourCfile.cpp` and it will compile your c++ code and produce a binary with your code output which you can get using : `./yourBinary` , (You can use Acode terminal plugin with Code runner to speed up this process) 

But if you are not familiar with terminal etc , then you should try something else like: Replit or Cxxdroid, etc..

**Reply by Medboy224** [reply]
Ok, I understand.  So it is not possible to launch my C++ source code directly in Acode like Cxxdroid does.  Wouldn't there be a possibility to do this directly in Acode?  and please can you tell me which programming languages ​​can be run directly in Acode

**Reply by bajrangCoder** [reply]
It's possible to run with plugins that will use termux in backend to run and You can run html, css and javascript directly and Also python with few limitations.

> Btw without plugins Acode is only a code editor and ide for html, css and javascript 

> [!Note]
> If you have little bit knowledge of terminals then Acode is good for you with Plugins, same as in vscode user use terminal to run different programming language, But on Mobile it's Little bit hard to create Terminal directly that's why many plugin rely on Termux (because it's matured enough)

**Reply by Medboy224** [reply]
I was able to install the AcodeX-terminal plugin and launch termux in the background with a server connected to Acode.  Now I can write my code in Acode, compile it via the plugin and run it, really thank you very much.  The only problem I'm having now is that Acode can't highlight errors in my C++ code.

**Comment by TGJASSIYT** [comment]
How to save all files at once?

**Reply by bajrangCoder** [reply]
Check command palette `save all changes` command

**Reply by TGJASSIYT** [reply]
> Check command palette `save all changes` command

Alright but is there any shortcut key to do that instead of using command palette

**Reply by bajrangCoder** [reply]
There is no any shortcut key but you can set it in keybindings file.
For eg:

```json
"saveAllChanges": {
    "key": "Ctrl-Shift-Alt-S"
}
```

> Note: key shouldn't clash with other

**Reply by TGJASSIYT** [reply]
This actually does save all files but there must be an option to save multiple of them at once, it commits two times for two files in Github Repository

**Reply by bajrangCoder** [reply]
But you had asked about "How to save all file once", this command is to save all unsaved buffer opened in editor.
Btw I don't know about GitHub plugin, it maybe GitHub plugin issue, not the Acode itself

**Comment by TGJASSIYT** [comment]
I have a suggestion: files should be cached to work offline and sync with GitHub when internet is available and user syncs so. This way, we can code and see results without constantly reloading files online, which can be slow. When the internet is available, we can save changes made to the cached files or sync them again which only rewrites the files that are different and does not load everything again, with a warning to prevent losing new changes by overwriting them with old data from the repository. Questions are welcome for this suggestion if something is not clear.

**Reply by bajrangCoder** [reply]
Hmm , GitHub plugin is not usable on large scale, as it is made for small works and beginners but It can be improved and GitHub feature is now available as plugin so it's good to report it there.

Btw For your case, you need Git client and there is a Git plugin already available for Acode.
You can try that.

**Comment by JankleDev** [comment]
the existing github plugin is really not that efficient and easy to work
with, so i suggest that you use other apps like Termux or a github client
like GitSync. heres the app i currently use
https://play.google.com/store/apps/details?id=com.viscouspot.gitsync

**Reply by TGJASSIYT** [reply]
Seems exactly what i want but it has no option to run the code so umm what could be done

**Comment by TGJASSIYT** [comment]
Can someone make a plugin that detects variables in css files and shows what colors they actually have if available in the same profile, like hex code shows what color it is actually

**Comment by JankleDev** [comment]
Extra syntax highlight 2.0

**Comment by yazigegeda** [comment]
How to get code suggestions when writing C/C++ code in VS Code? I tried installing the plugins and configuring clangd, but it still doesn't provide code suggestions. Please help me.

**Comment by akunkosong** [comment]
Why cannot insatall kivy and or Kivymd

(apps) root@localhost ~/apps $ pip install kivy kivymd
Collecting kivy
  Downloading Kivy-2.3.1.tar.gz (24.0 MB)
     ━━━━━━━━━━━━━━ 24.0/24.0 MB 2.7 MB/s eta 0:00:00
  Installing build dependencies ... done
  ERROR: Error [Errno 2] No such file or directory: '/data/user/0/com.foxdebug.acode/cache/pip-install-fwn2q792/kivy_30624edd127c43dabc77aa7872fc84d9' while executing command Getting requirements to build wheel
  Getting requirements to build wheel ... error
ERROR: Could not install packages due to an OSError: [Errno 2] No such file or directory: '/data/user/0/com.foxdebug.acode/cache/pip-install-fwn2q792/kivy_30624edd127c43dabc77aa7872fc84d9'

Traceback (most recent call last):
  File "/usr/bin/pip", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/usr/lib/python3.12/site-packages/pip/_internal/cli/main.py", line 80, in main
    return command.main(cmd_args)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/pip/_internal/cli/base_command.py", line 156, in main
    with self.main_context():
         ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/contextlib.py", line 144, in __exit__
    next(self.gen)
  File "/usr/lib/python3.12/site-packages/pip/_internal/cli/command_context.py", line 19, in main_context
    with self._main_context:
         ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/contextlib.py", line 610, in __exit__
    raise exc_details[1]
  File "/usr/lib/python3.12/contextlib.py", line 595, in __exit__
    if cb(*exc_details):
       ^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/contextlib.py", line 144, in __exit__
 next(self.gen)
  File "/usr/lib/python3.12/site-packages/pip/_internal/operations/build/build_tracker.py", line 49, in get_build_tracker
    with BuildTracker(root) as tracker:
         ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/pip/_internal/operations/build/build_tracker.py", line 81, in __exit__
    self.cleanup()
  File "/usr/lib/python3.12/site-packages/pip/_internal/operations/build/build_tracker.py", line 125, in cleanup
    self.remove(req, key)
  File "/usr/lib/python3.12/site-packages/pip/_internal/operations/build/build_tracker.py", line 118, in remove
    os.unlink(self._entry_path(key))
FileNotFoundError: [Errno 2] No such file or directory: '/data/user/0/com.foxdebug.acode/cache/pip-build-tracker-a7t5uo4o/2f8041951800ec6a7b0e6fb07ded6d253584a11a2e54962805d33653'

**Reply by RohitKushvaha01** [reply]
Kivy is not supported in acode terminal

**Comment by somailawedamlivingstone-ctrl** [comment]
How to integrate ai in file
